src/messaging/ReliableMessageProtocolConfig.h

```diff
@@ -90,7 +90,7 @@ namespace Messaging {
  *
  */
 #ifndef CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE
-#ifdef PBUF_POOL_SIZE
+#if !LWIP_PBUF_FROM_CUSTOM_POOLS && PBUF_POOL_SIZE
```

Comment:
I would in general argue for this config to ALWAYS be set **per product**, and not try to make assumptions based on any particular usage of lwIP...

Reply:
It looks like there is a quite complex logic to choose the `CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE`. I'm not a fun of it.

Reply:
Yes, this logic is indeed looks like a puzzle, try to simply it as much as I can. 

I agree with @tcarmelveilleux there is no guarantee config LWIP_PBUF_FROM_CUSTOM_POOLS is always set for each product. so we just define CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE to CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS only if this config flag is defined and not set to 0.

Reply:
Changed #ifdef PBUF_POOL_SIZE to #elif PBUF_POOL_SIZE which should have similar effect as static_assert check